persist/fs/seek.go

```diff
@@ -278,51 +296,27 @@ func (s *seeker) readInfo(size int) error {
 	s.blockSize = time.Duration(info.BlockSize)
 	s.entries = int(info.Entries)
 	s.bloomFilterInfo = info.BloomFilter
+	s.summariesInfo = info.Summaries
 
 	return nil
 }
 
-func (s *seeker) readIndex(size int) error {
-	s.prepareUnreadBuf(size)
-	indexBytes := s.unreadBuf[:size]
-	n, err := s.indexFdWithDigest.ReadAllAndValidate(indexBytes, s.expectedIndexDigest)
+// SeekByID returns the data for the specified ID. An error will be returned if the
+// ID cannot be found.
+func (s *seeker) SeekByID(id ts.ID) (checked.Bytes, error) {
+	entry, err := s.SeekIndexEntry(id)
 	if err != nil {
-		return err
-	}
-
-	s.decoder.Reset(encoding.NewDecoderStream(s.unreadBuf[:n][:]))
-	if s.indexMap == nil {
-		s.indexMap = make(map[ts.Hash]indexMapEntry, s.entries)
-	}
-	// Read all entries of index
-	for read := 0; read < s.entries; read++ {
-		entry, err := s.decoder.DecodeIndexEntry()
-		if err != nil {
-			return err
-		}
-		s.indexMap[ts.HashFn(entry.ID)] = indexMapEntry{
-			size:     uint32(entry.Size),
-			checksum: uint32(entry.Checksum),
-			offset:   entry.Offset,
-		}
-
-		if s.keepIndexIDs {
-			entryID := append([]byte(nil), entry.ID...)
-			id := ts.BinaryID(checked.NewBytes(entryID, nil))
-			s.indexIDs = append(s.indexIDs, id)
-		}
+		return nil, err
 	}
 
-	return nil
+	return s.SeekByIndexEntry(entry)
 }
 
-func (s *seeker) Seek(id ts.ID) (checked.Bytes, error) {
-	entry, exists := s.indexMap[id.Hash()]
-	if !exists {
-		return nil, errSeekIDNotFound
-	}
-
-	_, err := s.dataFd.Seek(entry.offset, 0)
+// SeekByIndexEntry is similar to Seek, but uses the provided IndexEntry
+// instead of looking it up on its own. Useful in cases where you've already
+// obtained an entry and don't want to waste resources looking it up again.
+func (s *seeker) SeekByIndexEntry(entry IndexEntry) (checked.Bytes, error) {
+	_, err := s.dataFd.Seek(entry.Offset, 0)
```

Comment:
Hm, I thought we'd probably be using mmap for the data file too to reduce syscalls and reuse page caches (if some of them begin being read very frequently).  Thoughts?

Reply:
Spoke in person, decided to handle this in a subsequent P.R